src/main/scala/com/tuplejump/kafka/connect/cassandra/TaskConfig.scala

```diff
@@ -319,15 +324,32 @@ private[cassandra] object InternalConfig {
   sealed trait ClusterQueryOptions
 
   /** Settings related for individual queries, can be set per keyspace.table. */
-  final case class WriteOptions(consistency: ConsistencyLevel) extends ClusterQueryOptions
+  final case class WriteOptions(consistency: ConsistencyLevel,
+                                fieldMapping: Map[String, Any]) extends ClusterQueryOptions
 
   object WriteOptions {
 
-    def apply(config: Map[String,String]): WriteOptions =
+    def apply(config: Map[String, String]): WriteOptions = {
+      val fieldMapping = config.get(FieldMapping) match {
+        case Some(fieldMappingString) =>
+          JSON.parseFull(fieldMappingString) match {
+            case Some(data) => {
+              data match {
+                case map: Map[_, _] =>
+                  map.asInstanceOf[Map[String, Any]]
+                case other =>
+                  throw new DataException(s"Invalid value for fieldMappingString $fieldMappingString")
+              }
+            }
+            case None =>
+              throw new DataException(s"Unable to parse fieldMappingString $fieldMappingString")
+          }
+        case None => DefaultFieldMapping
+      }
       WriteOptions(config.valueOr[ConsistencyLevel](
-        SinkConsistency, toConsistency, DefaultSourceConsistency))
+        SinkConsistency, toConsistency, DefaultSourceConsistency), fieldMapping)
+    }
   }
```

Comment:
def apply(config: Map[String, String]): WriteOptions = {

      def toMap(s: String) = JSON.parseFull(s) collect {
        case data: Map[_, _] => data.asInstanceOf[Map[String, Any]]
      } getOrElse(throw new DataException(s"Field mapping type for '$s' is not supported."))

      WriteOptions(
        consistency = config.valueOr[ConsistencyLevel](
          SinkConsistency, toConsistency, DefaultSourceConsistency),
        fieldMapping = config.valueOr[Map[String, Any]](
          FieldMapping, toMap, DefaultFieldMapping))
    }